src/liballoc/vec.rs

```diff
@@ -2337,22 +2338,20 @@ impl<T> Iterator for IntoIter<T> {
         unsafe {
             if self.ptr as *const _ == self.end {
                 None
+            } else if mem::size_of::<T>() == 0 {
```

Comment:
Can this be left as it was, it's a bit clearer in terms of control flow previously

Reply:
I can revert it but I'm a little confused how `else { if` is more clear than `else if`; it just adds unnecessary indentation IMHO.